otter/test/models/test_cass_models.py

```diff
@@ -857,22 +875,16 @@ def test_modify_state_lock_not_acquired(self, mock_serial):
         ``modify_state`` raises error if lock is not acquired and does not
         do anything else
         """
-        self.lock.acquire.side_effect = \
-            lambda timeout: defer.fail(ValueError('a'))
+        self.acquire_call = (True, 10, Failure(ValueError("eh")))
 
         def modifier(group, state):
             raise
 
         self.group.view_state = mock.Mock(return_value=defer.succeed('state'))
-
         d = self.group.modify_state(modifier)
         self.failureResultOf(d, ValueError)
-
         self.assertEqual(self.connection.execute.call_count, 0)
-        self.kz_client.Lock.assert_called_once_with(
-            '/locks/' + self.group.uuid)
-        self.lock._acquire.assert_called_once_with(timeout=10)
-        self.assertEqual(self.lock.release.call_count, 0)
+        self.assertFalse(self.lock.acquired)
```

Comment:
Hmm.. interesting. I hadn't thought of it like this. My immediate reaction is that this seems to be over-engineering mainly because the public attribute is being used in test-facing code only. But I respect your experience to realize that you might have _actually_ faced this problem. I'll try to refactor this in this PR but if it seems to be taking more time than I would like, I'll create a tech-debt ticket instead. Thanks!